public_html/question2.js

```diff
@@ -0,0 +1,41 @@
+let n = prompt("Enter a number (n*n grid)");
+var c = ["R","B","Y","G"]; //create an array of colors
+
+function createMap(n, n) {
+    let map = [];
+    for(let x = 0; x < n; x++){
+        map[x] = [];
+        for(let y = 0; y < n; y++){
+            map[x][y] = {
+                color: c[Math.floor(Math.random() * c.length)], //randomly selects color from array c
+                val: Math.floor(Math.random() * 9) //randomly selects a value between 1-9 inclusive
```

Comment:
Generation of random integer is off by 1. This generates numbers between 0 and 8.